generated/nifake/tests/test_session.py

```diff
@@ -380,3 +380,19 @@ def test_init_with_options_nondefault(self):
         session = nifake.Session('FakeDevice', True, True, 'Some string')
         assert(session.vi == SESSION_NUM_FOR_TEST)
         self.patched_library.niFake_InitWithOptions.assert_called_once_with(b'FakeDevice', True, True, b'Some string', ANY)
+
+    def test_set_bool_attribute(self):
+        self.patched_library.niFake_SetAttributeViBoolean.side_effect = self.side_effects_helper.niFake_SetAttributeViBoolean
+        attribute_id = 1000000
+        attrib_bool = True
+        with nifake.Session('dev1') as session:
+            session.read_write_bool = attrib_bool
+            self.patched_library.niFake_SetAttributeViBoolean.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', attribute_id, attrib_bool)
+
+    def test_set_vi_string_attribute(self):
+        self.patched_library.niFake_SetAttributeViString.side_effect = self.side_effects_helper.niFake_SetAttributeViString
+        attribute_id = 1000002
+        attrib_string = 'This is test string'
+        with nifake.Session('dev1') as session:
+            session.read_write_string = attrib_string
+            self.patched_library.niFake_SetAttributeViString.assert_called_once_with(SESSION_NUM_FOR_TEST, b'', attribute_id, attrib_string)
```

Comment:
`assertion error value = AssertionError("Expected call: niFake_SetAttributeViString(42, b'', 1000002, 'This is test string')\nActual call: niFake_SetAttributeViString(42, b'', 1000002, b'This is test string')",) `

Should I use bytes(variable) in expected call ?

Reply:
Using bytes() will make the test pass.

Now whether that is correct or not, I am not sure. 

If the underlying driver doesn't care if it is a string or bytes in Python, then I think adding bytes() is an acceptable solution to this test.

If it does care, then we need to change things so that we pass in a string and not bytes, in which case you wouldn't need to change this.

Reply:
I think it's correct that the expectations for the ctypes mock is that it receives a buffer of bytes.